### PR TITLE
feat(ibm): add container_vpc_cluster

### PR DESCRIPTION
**Pull request by balazs-marjan**

### CODE REVIEW

**Review by hiltol — CHANGES_REQUESTED** [review]

File: internal/resources/ibm/container_vpc_cluster.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+		{Key: "isolation", Value: strPtr("public")},
+		{Key: "catalogRegion", Value: strPtr(r.Region)},
+		{Key: "operatingSystem", Value: strPtr(operatingSystem)},
+	}
+	if r.Entitlement {
+		attributeFilters = append(attributeFilters, &schema.AttributeFilter{
+			Key: "ocpIncluded", Value: strPtr("true"),
+		})
+	} else {
+		attributeFilters = append(attributeFilters, &schema.AttributeFilter{
```
[the remark this answers is not in the document]
Comment (balazs-marjan) [reply internal/resources/ibm/container_vpc_cluster.go:73]: If I don't pass it in, ocp and no ocp options. The backend will handle the null check, you can check the `filterToCondition` function

File: internal/resources/ibm/container_vpc_worker_pool.go [hunk 2]
```diff
@@ -20,37 +21,44 @@ import (
...
+	if isOpenshift {
+		operatingSystem = "REDHAT"
+	}
 	var attributeFilters = []*schema.AttributeFilter{
 		{Key: "provider", Value: strPtr("vpc-gen2")},
 		{Key: "flavor", Value: strPtr(r.Flavor)},
 		{Key: "serverType", Value: strPtr("virtual")},
 		{Key: "isolation", Value: strPtr("public")},
 		{Key: "catalogRegion", Value: strPtr(r.Region)},
+		{Key: "operatingSystem", Value: strPtr(operatingSystem)},
```
[the remark this answers is not in the document]
Comment (balazs-marjan) [reply internal/resources/ibm/container_vpc_worker_pool.go:61]: ok

File: internal/resources/ibm/ibm_is_vpc.go [hunk 3]
```diff
@@ -29,16 +33,19 @@ func (r *IbmIsVpc) PopulateUsage(u *schema.UsageData) {
 
 // @TODO: add/find a '$0' entry in pricing db to allow zero dollar cost components to show up in breakdown
 func (r *IbmIsVpc) vpcCostComponent() *schema.CostComponent {
+	var quantity *decimal.Decimal
+	if r.GigabyteTransmittedOutbounds != nil {
+		quantity = decimalPtr(decimal.NewFromFloat(*r.GigabyteTransmittedOutbounds))
+	}
 	return &schema.CostComponent{
-		Name:            fmt.Sprintf("VPC %s", r.Region),
-		Unit:            "Instance",
```
[the remark this answers is not in the document]
Comment (balazs-marjan) [reply internal/resources/ibm/ibm_is_vpc.go:34]: ok

File: internal/providers/terraform/ibm/testdata/ibm_is_vpc_test/ibm_is_vpc_test.golden [hunk 4]
```diff
@@ -1,10 +1,9 @@
 
- Name                Monthly Qty  Unit      Monthly Cost 
-                                                         
- ibm_is_vpc.testVpc                                      
- └─ VPC us-south               1  Instance         $0.00
```
[the remark this answers is not in the document]
Comment (balazs-marjan) [reply internal/providers/terraform/ibm/testdata/ibm_is_vpc_test/ibm_is_vpc_test.golden:5]: ok

File: internal/resources/ibm/container_vpc_cluster.go [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+	if isOpenshift {
+		operatingSystem = "REDHAT"
+	}
+	var attributeFilters = []*schema.AttributeFilter{
+		{Key: "provider", Value: strPtr("vpc-gen2")},
+		{Key: "flavor", Value: strPtr(r.Flavor)},
+		{Key: "serverType", Value: strPtr("virtual")},
+		{Key: "isolation", Value: strPtr("public")},
+		{Key: "catalogRegion", Value: strPtr(r.Region)},
+		{Key: "operatingSystem", Value: strPtr(operatingSystem)},
```
[the remark this answers is not in the document]
Comment (balazs-marjan) [reply internal/resources/ibm/container_vpc_cluster.go:66]: ok

File: internal/providers/terraform/ibm/testdata/ibm_is_vpc_test/ibm_is_vpc_test.golden (hunk 4 above)
[the remark this answers is not in the document]
Comment (balazs-marjan) [reply internal/providers/terraform/ibm/testdata/ibm_is_vpc_test/ibm_is_vpc_test.golden:5]: resolved in the other PR

**Review by sonkatomi — APPROVED** [review]

**Review by hiltol — APPROVED** [review]

**Pull request merged**